packages/material-ui/src/utils/integerPropType.test.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { expect } from 'chai';
+import { integerPropType } from '.';
+
+function callValidator(
```

Comment:
See https://github.com/mui-org/material-ui/blob/a628fc2ccb6a9ec85ce87de83acdd8d6bb3339e3/packages/material-ui-utils/src/chainPropTypes.test.ts#L20 for how we test our custom prop-types.

Reply:
But you can still create an abstraction:

https://github.com/mui-org/material-ui/blob/a628fc2ccb6a9ec85ce87de83acdd8d6bb3339e3/packages/material-ui-utils/src/elementAcceptingRef.test.tsx#L25

Reply:
@oliviertassinari  `PropTypes.resetWarningCache()` is available only in `production`. I've tested but it doesn't work if the `NODE_ENV` is not set to `production`. Should I push the changes in order to make it clear or do you have any suggestions?

An example of how I'm wrapping the tests:
```javascript
 beforeEach(() => {
    PropTypes.resetWarningCache();
  });

  function assertPasses({ props }, propName, required = false) {
    PropTypes.checkPropTypes(
      {
        [propName]: required ? integerPropType.isRequired : integerPropType,
      },
      props,
      location,
      componentName,
    );

    expect(() => {}).not.toErrorDev();
  }

  function assertFails({ props }, propName, required = false) {
    const propType = getTypeByValue(props[propName]);
    const errorMessage = `Warning: Failed  type: Invalid ${location} \`${propName}\` of type \`${propType}\` supplied to \`${componentName}\`, expected \`integer\`.`

    expect(() => {
      PropTypes.checkPropTypes(
        {
          [propName]: required ? integerPropType.isRequired : integerPropType,
        },
        props,
        location,
        componentName,
      );
    }).toErrorDev(errorMessage);
  }
```

Reply:
> PropTypes.resetWarningCache() is available only in production link

I read the opposite, only available in development.

Reply:
Yeah, I've seen the library, but on my workspace the index.d.ts of prop-types, I've found this:
```javascript
/**
 * Only available if NODE_ENV=production
 */
export function resetWarningCache(): void;
```
3 tests are failing (which are related to integerPropType), but this thing (` * Only available if NODE_ENV=production`) made me confused...

Reply:
To clarify it, the error message occurs after failing the test:
```sh
AssertionError: Could not match the following console.error calls. Make sure previous actions didn't call console.error by wrappin
g them in expect(() => {}).not.toErrorDev(): 
```

Reply:
@oliviertassinari I've found the mistake of mine :))